sklearn/decomposition/tests/test_factor_analysis.py

```diff
@@ -83,3 +93,32 @@ def test_factor_analysis():
         precision = fa.get_precision()
         assert_array_almost_equal(np.dot(cov, precision),
                                   np.eye(X.shape[1]), 12)
+
+
+@pytest.mark.parametrize("data_type, expected_type, noise_variance_init", [
+    (np.float32, np.float32, None),
+    (np.float64, np.float64, None),
+    (np.int32, np.float64, None),
+    (np.int64, np.float64, None),
+    (np.float32, np.float32, np.ones(n_features)),
+    (np.float64, np.float64, np.ones(n_features)),
+    (np.int32, np.float64, np.ones(n_features)),
+    (np.int64, np.float64, np.ones(n_features))
+])
+def test_lda_dtype_match(data_type, expected_type, noise_variance_init):
+    for method in ['randomized', 'lapack']:
+        clf = FactorAnalysis(n_components=n_components, svd_method=method,
+                             noise_variance_init=noise_variance_init)
+        clf.fit(X.astype(data_type))
+        assert clf.components_.dtype == expected_type
+
+
+@pytest.mark.parametrize("method", ['lapack', 'randomized'])
+def test_lda_numeric_consistency_float32_float64(method):
```

Comment:
Considering #13309 I tried running SVD in np.float64 by doing
```py
s, V, unexp_var = my_svd((X / (sqrt_psi * nsqrt)).astype(np.float64))
s, V, unexp_var = s.astype(X.dtype), V.astype(X.dtype), unexp_var.astype(X.dtype)
```
But it does not pass the numerical consistency test. There might other sources of numerical stability between types.